src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -650,15 +688,15 @@ def _push(
         self._steps.append((TokenType.OPER, ")"))
         self._steps.append((TokenType.OPER, oper))
 
-        # pylint: disable=protected-access
         if isinstance(other, (FormulaEngine, FormulaEngine3Phase)):
             self._steps.append((TokenType.COMPONENT_METRIC, other))
-        elif isinstance(other, (Quantity, float)):
+        elif isinstance(other, (Quantity, float, int)):
             match oper:
-                case "+" | "-":
+                case "+" | "-" | "max" | "min":
```

Comment:
is min and max really supposed to be a binary expression, such as `A min B` rather than `min(A, B)`? If so, why? :)

Reply:
I guess we're starting to see the postfix evaluation approach starting to show cracks.

Reply:
can you please give an example for the cracks?

Reply:
btw. it's implemented as A.max(B) ;)

Reply:
I guess this PR is the example.  As @christianparpart points out, `min` and `max` are not binary operators, but functions.  We do seem to be able to hack it into our postfix expression evaluator, but it is not ideal, and we still have to add tests and prove that it works in all cases, because this kind of thing can get very buggy very quickly.

I was thinking about an ast walker when doing the original implementation, but the requirements were so simple that it didn't make sense.  If we want to extend the possible operations further after this, we should replace the implementation first.

Reply:
There is no difference between operators and functions here. The + and * operators are simply functions mapping from float x float → float same as min and max.

Reply:
Whenever that topic might be prioritized, I'd love to implement the AST and related work

Reply:
> Whenever that topic might be prioritized, I'd love to implement the AST and related work
What do you mean? There is no problem here so far.